cmd/ontap-cvo.go

```diff
@@ -55,83 +55,69 @@ type s3KeysObj struct {
+	// right now this is the only place we will create the secret, so I will just have it in here
+	postResponseFormatted := &createUserResponse{} // must decode the []byte response into something i can mess with
+	// need to determine if this unmarshals / converts to the struct correctly
+	err := json.Unmarshal(response, &postResponseFormatted)
 	if err != nil {
-		klog.Infof("An Error Occured while getting netapp management information secret %v", err)
+		fmt.Println("Error in JSON unmarshalling from json marshalled object:", err)
 		return false
 	}
-
-	managementIP := string(secret.Data["management_ip"])
-	netappUser := string(secret.Data[filerStr+"_user"])
-	// netappUserAccess := string(secret.Data[filerStr+"_access"])
-	netappUserSecret := string(secret.Data[filerStr+"_secret"])
-
-	url := "https://" + managementIP + "/api/protocols/s3/services/" + filerUUID + "/users"
-	req, err := http.NewRequest("POST", url, bytes.NewBuffer(postBody))
-	req.Header.Set("Content-Type", "application/json")
-	// req.Header. // need to set other information
-	authorization := base64.StdEncoding.EncodeToString([]byte(netappUser + ":" + netappUserSecret)) // this must be confirmed)
-	req.Header.Set("Authorization", "Basic "+authorization)
-	// resp, err := http.Post(url, "application/json", bytes.NewBuffer(postBody)) // cant use this because http.post does not allow for additional headers
-	//Handle Error
-	/*if err != nil {
-		klog.Infof("An Error Occured while calling s3 services %v", err)
-	}*/
-	resp, err := http.DefaultClient.Do(req)
-	if err != nil {
-		klog.Fatalf("error sending and returning HTTP response  : %v", err)
-	}
-	// defer resp.Body.Close() // unsure if needed?
-
-	//post := &s3keys{}
-	post := &createUserResponse{}
-	derr := json.NewDecoder(resp.Body).Decode(post)
-	if derr != nil {
-		panic(derr)
-	}
-	if resp.StatusCode != http.StatusCreated {
-		panic(resp.Status)
-	}
-	// Now that we have the values for the keys put it into a secret in the namespace
+	// Create the secret
 	usersecret := &corev1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      filerStr + "-conn-secret", // change to be a const later or something
+			Name:      svmInfo.svmName + "-conn-secret", // change to be a const later or something
 			Namespace: namespaceStr,
 		},
 		Data: map[string][]byte{
-			// this [0] seems a bit suspect but we will see how it works for now I don't know
-			// I don't think the S3 account will be used multiple times or anything
-			"S3_ACCESS": []byte(post.records[0].AccessKey),
-			"S3_SECRET": []byte(post.records[0].SecretKey),
+			// Nothing else needs to be in here; as the S3_BUCKET value should be somewhere else.
+			// All S3 buckets under the same SVM use the same ACCESS and SECRET to access them
+			"S3_ACCESS": []byte(postResponseFormatted.records[0].AccessKey),
+			"S3_SECRET": []byte(postResponseFormatted.records[0].SecretKey),
+			"S3_URL":    []byte(svmInfo.svmUrl),
```

Comment:
S3 URL wont be in the secret. It will just be called by the injector when needed. This is to reduce the duplication of this information so that the source of truth will always be the master CM